core/execution/amm/pool.go

```diff
@@ -468,14 +521,10 @@ func (p *Pool) fairPrice() *num.Uint {
 // virtualBalances returns the pools x, y values where x is the balance in contracts and y is the balance in asset.
 func (p *Pool) virtualBalances(pos int64, ae *num.Uint, side types.Side) (num.Decimal, num.Decimal) {
 	switch {
-	case pos < 0:
-		return p.virtualBalancesShort(pos, ae)
-	case pos > 0:
-		return p.virtualBalancesLong(pos, ae)
-	case side == types.SideBuy:
+	case pos < 0, side == types.SideBuy:
```

Comment:
So it was like how it was before because only want to consider the side if position is `0`. If the position is non-zero we already know which curve to look at and we always want to use that one. When its `0` we use the incoming order side to work out which curve the AMM will move into, so we only want to do those cases after `pos > 0, pos <0` haven't been hit.

Reply:
I'm not sure if this is going to be functionally different, but will fix to be safe